src/ScottPlot4/ScottPlot/Plottable/FinancePlot.cs

```diff
@@ -294,9 +295,33 @@ private void RenderOhlc(PlotDimensions dims, Bitmap bmp, bool lowQuality)
             if (N >= OHLCs.Count)
                 throw new ArgumentException("can not analyze more points than are available in the OHLCs");
 
-            double[] xs = OHLCs.Skip(N).Select(x => x.DateTime.ToOADate()).ToArray();
-            (var sma, var lower, var upper) = Statistics.Finance.Bollinger(OHLCs.ToArray(), N);
+            var sortedOHLCs = GetSortedOHLCs();
+            double[] xs = sortedOHLCs.Skip(N).Select(x => x.DateTime.ToOADate()).ToArray();
+            (var sma, var lower, var upper) = Statistics.Finance.Bollinger(sortedOHLCs, N);
             return (xs, sma, lower, upper);
         }
+
+        private OHLC[] GetSortedOHLCs()
+        {
+            if (OHLCsAreSorted())
+            {
+                return OHLCs.ToArray();
```

Comment:
If they're already sorted I don't think we need to copy it? We can just return `OHLCs` and we'll save a copy?

Reply:
Good call that is simpler, fixed